clarity/src/vm/database/clarity_db.rs

```diff
@@ -1075,35 +1075,41 @@ impl<'a> ClarityDatabase<'a> {
     /// block height (i.e. that returned by `get_index_block_header_hash` for
     /// `get_current_block_height`).
     pub fn get_current_burnchain_block_height(&mut self) -> Result<u32> {
-        let cur_stacks_height = self.store.get_current_block_height();
+        let epoch = self.get_clarity_epoch_version()?;
+        match epoch {
+            // Special case to preserve possibly incorrect behavior (inside at-block) in Epoch 3.0
+            StacksEpochId::Epoch30 => {
```

Comment:
```suggestion
            StacksEpochId::Epoch30 | StacksEpochId::Epoch31  => {
```

(Seems obvious, but let's not forget about it)

Reply:
That's why we avoid using the `_` case in stacks-core... we can't forget about it. May want to practice the same here.

Reply:
I think when I created this PR I was hoping to get it into Epoch 3.1, but that opportunity has passed

Reply:
> That's why we avoid using the _ case in stacks-core

Oh right, there should be a clippy rule about that but I can't find

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#wildcard_enum_match_arm

Reply:
But it's allowed by default and can't be set in clippy.toml

Reply:
Ah ok. There were some arguments made about not applying that rule any way, but in most cases, we should manually follow that rule.